Description: Upstream changes introduced in version 0.0.1+svn20110409.0930-2yavdr1~natty
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 eventlircd (0.0.1+svn20110409.0930-2yavdr1~natty) natty; urgency=low
 .
   * added missing option -R to getopt_long
 .
 The person named in the Author field signed this changelog entry.
Author: Gerald Dachs <gda@dachsweb.de>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- eventlircd-0.0.1+svn20110409.0930.orig/src/main.c
+++ eventlircd-0.0.1+svn20110409.0930/src/main.c
@@ -73,7 +73,7 @@ int main(int argc,char **argv)
 
     openlog(progname, LOG_CONS | LOG_PERROR | LOG_PID, LOG_DAEMON);
 
-    while((opt = getopt_long(argc, argv, "hVvfe:s:m:r:", longopts, NULL)) != -1)
+    while((opt = getopt_long(argc, argv, "hVvfe:s:m:r:R", longopts, NULL)) != -1)
     {
         switch(opt)
         {
--- /dev/null
+++ eventlircd-0.0.1+svn20110409.0930/src/event_name_to_code.h
@@ -0,0 +1,591 @@
+#ifndef _EVENTLIRCD_EVENT_NAME_TO_CODE_H_
+#define _EVENTLIRCD_EVENT_NAME_TO_CODE_H_ 1
+#include <///usr/include/linux/input.h>
+#include <linux/types.h>
+static const struct {
+	const char *name;
+	__u16 type;
+	__u16 code;
+} event_name_to_code[] = {
+	{ "SYN_REPORT"              , EV_SYN , SYN_REPORT              },
+	{ "SYN_CONFIG"              , EV_SYN , SYN_CONFIG              },
+	{ "SYN_MT_REPORT"           , EV_SYN , SYN_MT_REPORT           },
+	{ "KEY_ESC"                 , EV_KEY , KEY_ESC                 },
+	{ "KEY_1"                   , EV_KEY , KEY_1                   },
+	{ "KEY_2"                   , EV_KEY , KEY_2                   },
+	{ "KEY_3"                   , EV_KEY , KEY_3                   },
+	{ "KEY_4"                   , EV_KEY , KEY_4                   },
+	{ "KEY_5"                   , EV_KEY , KEY_5                   },
+	{ "KEY_6"                   , EV_KEY , KEY_6                   },
+	{ "KEY_7"                   , EV_KEY , KEY_7                   },
+	{ "KEY_8"                   , EV_KEY , KEY_8                   },
+	{ "KEY_9"                   , EV_KEY , KEY_9                   },
+	{ "KEY_0"                   , EV_KEY , KEY_0                   },
+	{ "KEY_MINUS"               , EV_KEY , KEY_MINUS               },
+	{ "KEY_EQUAL"               , EV_KEY , KEY_EQUAL               },
+	{ "KEY_BACKSPACE"           , EV_KEY , KEY_BACKSPACE           },
+	{ "KEY_TAB"                 , EV_KEY , KEY_TAB                 },
+	{ "KEY_Q"                   , EV_KEY , KEY_Q                   },
+	{ "KEY_W"                   , EV_KEY , KEY_W                   },
+	{ "KEY_E"                   , EV_KEY , KEY_E                   },
+	{ "KEY_R"                   , EV_KEY , KEY_R                   },
+	{ "KEY_T"                   , EV_KEY , KEY_T                   },
+	{ "KEY_Y"                   , EV_KEY , KEY_Y                   },
+	{ "KEY_U"                   , EV_KEY , KEY_U                   },
+	{ "KEY_I"                   , EV_KEY , KEY_I                   },
+	{ "KEY_O"                   , EV_KEY , KEY_O                   },
+	{ "KEY_P"                   , EV_KEY , KEY_P                   },
+	{ "KEY_LEFTBRACE"           , EV_KEY , KEY_LEFTBRACE           },
+	{ "KEY_RIGHTBRACE"          , EV_KEY , KEY_RIGHTBRACE          },
+	{ "KEY_ENTER"               , EV_KEY , KEY_ENTER               },
+	{ "KEY_LEFTCTRL"            , EV_KEY , KEY_LEFTCTRL            },
+	{ "KEY_A"                   , EV_KEY , KEY_A                   },
+	{ "KEY_S"                   , EV_KEY , KEY_S                   },
+	{ "KEY_D"                   , EV_KEY , KEY_D                   },
+	{ "KEY_F"                   , EV_KEY , KEY_F                   },
+	{ "KEY_G"                   , EV_KEY , KEY_G                   },
+	{ "KEY_H"                   , EV_KEY , KEY_H                   },
+	{ "KEY_J"                   , EV_KEY , KEY_J                   },
+	{ "KEY_K"                   , EV_KEY , KEY_K                   },
+	{ "KEY_L"                   , EV_KEY , KEY_L                   },
+	{ "KEY_SEMICOLON"           , EV_KEY , KEY_SEMICOLON           },
+	{ "KEY_APOSTROPHE"          , EV_KEY , KEY_APOSTROPHE          },
+	{ "KEY_GRAVE"               , EV_KEY , KEY_GRAVE               },
+	{ "KEY_LEFTSHIFT"           , EV_KEY , KEY_LEFTSHIFT           },
+	{ "KEY_BACKSLASH"           , EV_KEY , KEY_BACKSLASH           },
+	{ "KEY_Z"                   , EV_KEY , KEY_Z                   },
+	{ "KEY_X"                   , EV_KEY , KEY_X                   },
+	{ "KEY_C"                   , EV_KEY , KEY_C                   },
+	{ "KEY_V"                   , EV_KEY , KEY_V                   },
+	{ "KEY_B"                   , EV_KEY , KEY_B                   },
+	{ "KEY_N"                   , EV_KEY , KEY_N                   },
+	{ "KEY_M"                   , EV_KEY , KEY_M                   },
+	{ "KEY_COMMA"               , EV_KEY , KEY_COMMA               },
+	{ "KEY_DOT"                 , EV_KEY , KEY_DOT                 },
+	{ "KEY_SLASH"               , EV_KEY , KEY_SLASH               },
+	{ "KEY_RIGHTSHIFT"          , EV_KEY , KEY_RIGHTSHIFT          },
+	{ "KEY_KPASTERISK"          , EV_KEY , KEY_KPASTERISK          },
+	{ "KEY_LEFTALT"             , EV_KEY , KEY_LEFTALT             },
+	{ "KEY_SPACE"               , EV_KEY , KEY_SPACE               },
+	{ "KEY_CAPSLOCK"            , EV_KEY , KEY_CAPSLOCK            },
+	{ "KEY_F1"                  , EV_KEY , KEY_F1                  },
+	{ "KEY_F2"                  , EV_KEY , KEY_F2                  },
+	{ "KEY_F3"                  , EV_KEY , KEY_F3                  },
+	{ "KEY_F4"                  , EV_KEY , KEY_F4                  },
+	{ "KEY_F5"                  , EV_KEY , KEY_F5                  },
+	{ "KEY_F6"                  , EV_KEY , KEY_F6                  },
+	{ "KEY_F7"                  , EV_KEY , KEY_F7                  },
+	{ "KEY_F8"                  , EV_KEY , KEY_F8                  },
+	{ "KEY_F9"                  , EV_KEY , KEY_F9                  },
+	{ "KEY_F10"                 , EV_KEY , KEY_F10                 },
+	{ "KEY_NUMLOCK"             , EV_KEY , KEY_NUMLOCK             },
+	{ "KEY_SCROLLLOCK"          , EV_KEY , KEY_SCROLLLOCK          },
+	{ "KEY_KP7"                 , EV_KEY , KEY_KP7                 },
+	{ "KEY_KP8"                 , EV_KEY , KEY_KP8                 },
+	{ "KEY_KP9"                 , EV_KEY , KEY_KP9                 },
+	{ "KEY_KPMINUS"             , EV_KEY , KEY_KPMINUS             },
+	{ "KEY_KP4"                 , EV_KEY , KEY_KP4                 },
+	{ "KEY_KP5"                 , EV_KEY , KEY_KP5                 },
+	{ "KEY_KP6"                 , EV_KEY , KEY_KP6                 },
+	{ "KEY_KPPLUS"              , EV_KEY , KEY_KPPLUS              },
+	{ "KEY_KP1"                 , EV_KEY , KEY_KP1                 },
+	{ "KEY_KP2"                 , EV_KEY , KEY_KP2                 },
+	{ "KEY_KP3"                 , EV_KEY , KEY_KP3                 },
+	{ "KEY_KP0"                 , EV_KEY , KEY_KP0                 },
+	{ "KEY_KPDOT"               , EV_KEY , KEY_KPDOT               },
+	{ "KEY_ZENKAKUHANKAKU"      , EV_KEY , KEY_ZENKAKUHANKAKU      },
+	{ "KEY_102ND"               , EV_KEY , KEY_102ND               },
+	{ "KEY_F11"                 , EV_KEY , KEY_F11                 },
+	{ "KEY_F12"                 , EV_KEY , KEY_F12                 },
+	{ "KEY_RO"                  , EV_KEY , KEY_RO                  },
+	{ "KEY_KATAKANA"            , EV_KEY , KEY_KATAKANA            },
+	{ "KEY_HIRAGANA"            , EV_KEY , KEY_HIRAGANA            },
+	{ "KEY_HENKAN"              , EV_KEY , KEY_HENKAN              },
+	{ "KEY_KATAKANAHIRAGANA"    , EV_KEY , KEY_KATAKANAHIRAGANA    },
+	{ "KEY_MUHENKAN"            , EV_KEY , KEY_MUHENKAN            },
+	{ "KEY_KPJPCOMMA"           , EV_KEY , KEY_KPJPCOMMA           },
+	{ "KEY_KPENTER"             , EV_KEY , KEY_KPENTER             },
+	{ "KEY_RIGHTCTRL"           , EV_KEY , KEY_RIGHTCTRL           },
+	{ "KEY_KPSLASH"             , EV_KEY , KEY_KPSLASH             },
+	{ "KEY_SYSRQ"               , EV_KEY , KEY_SYSRQ               },
+	{ "KEY_RIGHTALT"            , EV_KEY , KEY_RIGHTALT            },
+	{ "KEY_LINEFEED"            , EV_KEY , KEY_LINEFEED            },
+	{ "KEY_HOME"                , EV_KEY , KEY_HOME                },
+	{ "KEY_UP"                  , EV_KEY , KEY_UP                  },
+	{ "KEY_PAGEUP"              , EV_KEY , KEY_PAGEUP              },
+	{ "KEY_LEFT"                , EV_KEY , KEY_LEFT                },
+	{ "KEY_RIGHT"               , EV_KEY , KEY_RIGHT               },
+	{ "KEY_END"                 , EV_KEY , KEY_END                 },
+	{ "KEY_DOWN"                , EV_KEY , KEY_DOWN                },
+	{ "KEY_PAGEDOWN"            , EV_KEY , KEY_PAGEDOWN            },
+	{ "KEY_INSERT"              , EV_KEY , KEY_INSERT              },
+	{ "KEY_DELETE"              , EV_KEY , KEY_DELETE              },
+	{ "KEY_MACRO"               , EV_KEY , KEY_MACRO               },
+	{ "KEY_MUTE"                , EV_KEY , KEY_MUTE                },
+	{ "KEY_VOLUMEDOWN"          , EV_KEY , KEY_VOLUMEDOWN          },
+	{ "KEY_VOLUMEUP"            , EV_KEY , KEY_VOLUMEUP            },
+	{ "KEY_POWER"               , EV_KEY , KEY_POWER               },
+	{ "KEY_KPEQUAL"             , EV_KEY , KEY_KPEQUAL             },
+	{ "KEY_KPPLUSMINUS"         , EV_KEY , KEY_KPPLUSMINUS         },
+	{ "KEY_PAUSE"               , EV_KEY , KEY_PAUSE               },
+	{ "KEY_SCALE"               , EV_KEY , KEY_SCALE               },
+	{ "KEY_KPCOMMA"             , EV_KEY , KEY_KPCOMMA             },
+	{ "KEY_HANGEUL"             , EV_KEY , KEY_HANGEUL             },
+	{ "KEY_HANGUEL"             , EV_KEY , KEY_HANGUEL             },
+	{ "KEY_HANJA"               , EV_KEY , KEY_HANJA               },
+	{ "KEY_YEN"                 , EV_KEY , KEY_YEN                 },
+	{ "KEY_LEFTMETA"            , EV_KEY , KEY_LEFTMETA            },
+	{ "KEY_RIGHTMETA"           , EV_KEY , KEY_RIGHTMETA           },
+	{ "KEY_COMPOSE"             , EV_KEY , KEY_COMPOSE             },
+	{ "KEY_STOP"                , EV_KEY , KEY_STOP                },
+	{ "KEY_AGAIN"               , EV_KEY , KEY_AGAIN               },
+	{ "KEY_PROPS"               , EV_KEY , KEY_PROPS               },
+	{ "KEY_UNDO"                , EV_KEY , KEY_UNDO                },
+	{ "KEY_FRONT"               , EV_KEY , KEY_FRONT               },
+	{ "KEY_COPY"                , EV_KEY , KEY_COPY                },
+	{ "KEY_OPEN"                , EV_KEY , KEY_OPEN                },
+	{ "KEY_PASTE"               , EV_KEY , KEY_PASTE               },
+	{ "KEY_FIND"                , EV_KEY , KEY_FIND                },
+	{ "KEY_CUT"                 , EV_KEY , KEY_CUT                 },
+	{ "KEY_HELP"                , EV_KEY , KEY_HELP                },
+	{ "KEY_MENU"                , EV_KEY , KEY_MENU                },
+	{ "KEY_CALC"                , EV_KEY , KEY_CALC                },
+	{ "KEY_SETUP"               , EV_KEY , KEY_SETUP               },
+	{ "KEY_SLEEP"               , EV_KEY , KEY_SLEEP               },
+	{ "KEY_WAKEUP"              , EV_KEY , KEY_WAKEUP              },
+	{ "KEY_FILE"                , EV_KEY , KEY_FILE                },
+	{ "KEY_SENDFILE"            , EV_KEY , KEY_SENDFILE            },
+	{ "KEY_DELETEFILE"          , EV_KEY , KEY_DELETEFILE          },
+	{ "KEY_XFER"                , EV_KEY , KEY_XFER                },
+	{ "KEY_PROG1"               , EV_KEY , KEY_PROG1               },
+	{ "KEY_PROG2"               , EV_KEY , KEY_PROG2               },
+	{ "KEY_WWW"                 , EV_KEY , KEY_WWW                 },
+	{ "KEY_MSDOS"               , EV_KEY , KEY_MSDOS               },
+	{ "KEY_COFFEE"              , EV_KEY , KEY_COFFEE              },
+	{ "KEY_SCREENLOCK"          , EV_KEY , KEY_SCREENLOCK          },
+	{ "KEY_DIRECTION"           , EV_KEY , KEY_DIRECTION           },
+	{ "KEY_CYCLEWINDOWS"        , EV_KEY , KEY_CYCLEWINDOWS        },
+	{ "KEY_MAIL"                , EV_KEY , KEY_MAIL                },
+	{ "KEY_BOOKMARKS"           , EV_KEY , KEY_BOOKMARKS           },
+	{ "KEY_COMPUTER"            , EV_KEY , KEY_COMPUTER            },
+	{ "KEY_BACK"                , EV_KEY , KEY_BACK                },
+	{ "KEY_FORWARD"             , EV_KEY , KEY_FORWARD             },
+	{ "KEY_CLOSECD"             , EV_KEY , KEY_CLOSECD             },
+	{ "KEY_EJECTCD"             , EV_KEY , KEY_EJECTCD             },
+	{ "KEY_EJECTCLOSECD"        , EV_KEY , KEY_EJECTCLOSECD        },
+	{ "KEY_NEXTSONG"            , EV_KEY , KEY_NEXTSONG            },
+	{ "KEY_PLAYPAUSE"           , EV_KEY , KEY_PLAYPAUSE           },
+	{ "KEY_PREVIOUSSONG"        , EV_KEY , KEY_PREVIOUSSONG        },
+	{ "KEY_STOPCD"              , EV_KEY , KEY_STOPCD              },
+	{ "KEY_RECORD"              , EV_KEY , KEY_RECORD              },
+	{ "KEY_REWIND"              , EV_KEY , KEY_REWIND              },
+	{ "KEY_PHONE"               , EV_KEY , KEY_PHONE               },
+	{ "KEY_ISO"                 , EV_KEY , KEY_ISO                 },
+	{ "KEY_CONFIG"              , EV_KEY , KEY_CONFIG              },
+	{ "KEY_HOMEPAGE"            , EV_KEY , KEY_HOMEPAGE            },
+	{ "KEY_REFRESH"             , EV_KEY , KEY_REFRESH             },
+	{ "KEY_EXIT"                , EV_KEY , KEY_EXIT                },
+	{ "KEY_MOVE"                , EV_KEY , KEY_MOVE                },
+	{ "KEY_EDIT"                , EV_KEY , KEY_EDIT                },
+	{ "KEY_SCROLLUP"            , EV_KEY , KEY_SCROLLUP            },
+	{ "KEY_SCROLLDOWN"          , EV_KEY , KEY_SCROLLDOWN          },
+	{ "KEY_KPLEFTPAREN"         , EV_KEY , KEY_KPLEFTPAREN         },
+	{ "KEY_KPRIGHTPAREN"        , EV_KEY , KEY_KPRIGHTPAREN        },
+	{ "KEY_NEW"                 , EV_KEY , KEY_NEW                 },
+	{ "KEY_REDO"                , EV_KEY , KEY_REDO                },
+	{ "KEY_F13"                 , EV_KEY , KEY_F13                 },
+	{ "KEY_F14"                 , EV_KEY , KEY_F14                 },
+	{ "KEY_F15"                 , EV_KEY , KEY_F15                 },
+	{ "KEY_F16"                 , EV_KEY , KEY_F16                 },
+	{ "KEY_F17"                 , EV_KEY , KEY_F17                 },
+	{ "KEY_F18"                 , EV_KEY , KEY_F18                 },
+	{ "KEY_F19"                 , EV_KEY , KEY_F19                 },
+	{ "KEY_F20"                 , EV_KEY , KEY_F20                 },
+	{ "KEY_F21"                 , EV_KEY , KEY_F21                 },
+	{ "KEY_F22"                 , EV_KEY , KEY_F22                 },
+	{ "KEY_F23"                 , EV_KEY , KEY_F23                 },
+	{ "KEY_F24"                 , EV_KEY , KEY_F24                 },
+	{ "KEY_PLAYCD"              , EV_KEY , KEY_PLAYCD              },
+	{ "KEY_PAUSECD"             , EV_KEY , KEY_PAUSECD             },
+	{ "KEY_PROG3"               , EV_KEY , KEY_PROG3               },
+	{ "KEY_PROG4"               , EV_KEY , KEY_PROG4               },
+	{ "KEY_DASHBOARD"           , EV_KEY , KEY_DASHBOARD           },
+	{ "KEY_SUSPEND"             , EV_KEY , KEY_SUSPEND             },
+	{ "KEY_CLOSE"               , EV_KEY , KEY_CLOSE               },
+	{ "KEY_PLAY"                , EV_KEY , KEY_PLAY                },
+	{ "KEY_FASTFORWARD"         , EV_KEY , KEY_FASTFORWARD         },
+	{ "KEY_BASSBOOST"           , EV_KEY , KEY_BASSBOOST           },
+	{ "KEY_PRINT"               , EV_KEY , KEY_PRINT               },
+	{ "KEY_HP"                  , EV_KEY , KEY_HP                  },
+	{ "KEY_CAMERA"              , EV_KEY , KEY_CAMERA              },
+	{ "KEY_SOUND"               , EV_KEY , KEY_SOUND               },
+	{ "KEY_QUESTION"            , EV_KEY , KEY_QUESTION            },
+	{ "KEY_EMAIL"               , EV_KEY , KEY_EMAIL               },
+	{ "KEY_CHAT"                , EV_KEY , KEY_CHAT                },
+	{ "KEY_SEARCH"              , EV_KEY , KEY_SEARCH              },
+	{ "KEY_CONNECT"             , EV_KEY , KEY_CONNECT             },
+	{ "KEY_FINANCE"             , EV_KEY , KEY_FINANCE             },
+	{ "KEY_SPORT"               , EV_KEY , KEY_SPORT               },
+	{ "KEY_SHOP"                , EV_KEY , KEY_SHOP                },
+	{ "KEY_ALTERASE"            , EV_KEY , KEY_ALTERASE            },
+	{ "KEY_CANCEL"              , EV_KEY , KEY_CANCEL              },
+	{ "KEY_BRIGHTNESSDOWN"      , EV_KEY , KEY_BRIGHTNESSDOWN      },
+	{ "KEY_BRIGHTNESSUP"        , EV_KEY , KEY_BRIGHTNESSUP        },
+	{ "KEY_MEDIA"               , EV_KEY , KEY_MEDIA               },
+	{ "KEY_SWITCHVIDEOMODE"     , EV_KEY , KEY_SWITCHVIDEOMODE     },
+	{ "KEY_KBDILLUMTOGGLE"      , EV_KEY , KEY_KBDILLUMTOGGLE      },
+	{ "KEY_KBDILLUMDOWN"        , EV_KEY , KEY_KBDILLUMDOWN        },
+	{ "KEY_KBDILLUMUP"          , EV_KEY , KEY_KBDILLUMUP          },
+	{ "KEY_SEND"                , EV_KEY , KEY_SEND                },
+	{ "KEY_REPLY"               , EV_KEY , KEY_REPLY               },
+	{ "KEY_FORWARDMAIL"         , EV_KEY , KEY_FORWARDMAIL         },
+	{ "KEY_SAVE"                , EV_KEY , KEY_SAVE                },
+	{ "KEY_DOCUMENTS"           , EV_KEY , KEY_DOCUMENTS           },
+	{ "KEY_BATTERY"             , EV_KEY , KEY_BATTERY             },
+	{ "KEY_BLUETOOTH"           , EV_KEY , KEY_BLUETOOTH           },
+	{ "KEY_WLAN"                , EV_KEY , KEY_WLAN                },
+	{ "KEY_UWB"                 , EV_KEY , KEY_UWB                 },
+	{ "KEY_UNKNOWN"             , EV_KEY , KEY_UNKNOWN             },
+	{ "KEY_VIDEO_NEXT"          , EV_KEY , KEY_VIDEO_NEXT          },
+	{ "KEY_VIDEO_PREV"          , EV_KEY , KEY_VIDEO_PREV          },
+	{ "KEY_BRIGHTNESS_CYCLE"    , EV_KEY , KEY_BRIGHTNESS_CYCLE    },
+	{ "KEY_BRIGHTNESS_ZERO"     , EV_KEY , KEY_BRIGHTNESS_ZERO     },
+	{ "KEY_DISPLAY_OFF"         , EV_KEY , KEY_DISPLAY_OFF         },
+	{ "KEY_WIMAX"               , EV_KEY , KEY_WIMAX               },
+	{ "KEY_RFKILL"              , EV_KEY , KEY_RFKILL              },
+	{ "BTN_MISC"                , EV_KEY , BTN_MISC                },
+	{ "BTN_0"                   , EV_KEY , BTN_0                   },
+	{ "BTN_1"                   , EV_KEY , BTN_1                   },
+	{ "BTN_2"                   , EV_KEY , BTN_2                   },
+	{ "BTN_3"                   , EV_KEY , BTN_3                   },
+	{ "BTN_4"                   , EV_KEY , BTN_4                   },
+	{ "BTN_5"                   , EV_KEY , BTN_5                   },
+	{ "BTN_6"                   , EV_KEY , BTN_6                   },
+	{ "BTN_7"                   , EV_KEY , BTN_7                   },
+	{ "BTN_8"                   , EV_KEY , BTN_8                   },
+	{ "BTN_9"                   , EV_KEY , BTN_9                   },
+	{ "BTN_MOUSE"               , EV_KEY , BTN_MOUSE               },
+	{ "BTN_LEFT"                , EV_KEY , BTN_LEFT                },
+	{ "BTN_RIGHT"               , EV_KEY , BTN_RIGHT               },
+	{ "BTN_MIDDLE"              , EV_KEY , BTN_MIDDLE              },
+	{ "BTN_SIDE"                , EV_KEY , BTN_SIDE                },
+	{ "BTN_EXTRA"               , EV_KEY , BTN_EXTRA               },
+	{ "BTN_FORWARD"             , EV_KEY , BTN_FORWARD             },
+	{ "BTN_BACK"                , EV_KEY , BTN_BACK                },
+	{ "BTN_TASK"                , EV_KEY , BTN_TASK                },
+	{ "BTN_JOYSTICK"            , EV_KEY , BTN_JOYSTICK            },
+	{ "BTN_TRIGGER"             , EV_KEY , BTN_TRIGGER             },
+	{ "BTN_THUMB"               , EV_KEY , BTN_THUMB               },
+	{ "BTN_THUMB2"              , EV_KEY , BTN_THUMB2              },
+	{ "BTN_TOP"                 , EV_KEY , BTN_TOP                 },
+	{ "BTN_TOP2"                , EV_KEY , BTN_TOP2                },
+	{ "BTN_PINKIE"              , EV_KEY , BTN_PINKIE              },
+	{ "BTN_BASE"                , EV_KEY , BTN_BASE                },
+	{ "BTN_BASE2"               , EV_KEY , BTN_BASE2               },
+	{ "BTN_BASE3"               , EV_KEY , BTN_BASE3               },
+	{ "BTN_BASE4"               , EV_KEY , BTN_BASE4               },
+	{ "BTN_BASE5"               , EV_KEY , BTN_BASE5               },
+	{ "BTN_BASE6"               , EV_KEY , BTN_BASE6               },
+	{ "BTN_DEAD"                , EV_KEY , BTN_DEAD                },
+	{ "BTN_GAMEPAD"             , EV_KEY , BTN_GAMEPAD             },
+	{ "BTN_A"                   , EV_KEY , BTN_A                   },
+	{ "BTN_B"                   , EV_KEY , BTN_B                   },
+	{ "BTN_C"                   , EV_KEY , BTN_C                   },
+	{ "BTN_X"                   , EV_KEY , BTN_X                   },
+	{ "BTN_Y"                   , EV_KEY , BTN_Y                   },
+	{ "BTN_Z"                   , EV_KEY , BTN_Z                   },
+	{ "BTN_TL"                  , EV_KEY , BTN_TL                  },
+	{ "BTN_TR"                  , EV_KEY , BTN_TR                  },
+	{ "BTN_TL2"                 , EV_KEY , BTN_TL2                 },
+	{ "BTN_TR2"                 , EV_KEY , BTN_TR2                 },
+	{ "BTN_SELECT"              , EV_KEY , BTN_SELECT              },
+	{ "BTN_START"               , EV_KEY , BTN_START               },
+	{ "BTN_MODE"                , EV_KEY , BTN_MODE                },
+	{ "BTN_THUMBL"              , EV_KEY , BTN_THUMBL              },
+	{ "BTN_THUMBR"              , EV_KEY , BTN_THUMBR              },
+	{ "BTN_DIGI"                , EV_KEY , BTN_DIGI                },
+	{ "BTN_TOOL_PEN"            , EV_KEY , BTN_TOOL_PEN            },
+	{ "BTN_TOOL_RUBBER"         , EV_KEY , BTN_TOOL_RUBBER         },
+	{ "BTN_TOOL_BRUSH"          , EV_KEY , BTN_TOOL_BRUSH          },
+	{ "BTN_TOOL_PENCIL"         , EV_KEY , BTN_TOOL_PENCIL         },
+	{ "BTN_TOOL_AIRBRUSH"       , EV_KEY , BTN_TOOL_AIRBRUSH       },
+	{ "BTN_TOOL_FINGER"         , EV_KEY , BTN_TOOL_FINGER         },
+	{ "BTN_TOOL_MOUSE"          , EV_KEY , BTN_TOOL_MOUSE          },
+	{ "BTN_TOOL_LENS"           , EV_KEY , BTN_TOOL_LENS           },
+	{ "BTN_TOUCH"               , EV_KEY , BTN_TOUCH               },
+	{ "BTN_STYLUS"              , EV_KEY , BTN_STYLUS              },
+	{ "BTN_STYLUS2"             , EV_KEY , BTN_STYLUS2             },
+	{ "BTN_TOOL_DOUBLETAP"      , EV_KEY , BTN_TOOL_DOUBLETAP      },
+	{ "BTN_TOOL_TRIPLETAP"      , EV_KEY , BTN_TOOL_TRIPLETAP      },
+	{ "BTN_TOOL_QUADTAP"        , EV_KEY , BTN_TOOL_QUADTAP        },
+	{ "BTN_WHEEL"               , EV_KEY , BTN_WHEEL               },
+	{ "BTN_GEAR_DOWN"           , EV_KEY , BTN_GEAR_DOWN           },
+	{ "BTN_GEAR_UP"             , EV_KEY , BTN_GEAR_UP             },
+	{ "KEY_OK"                  , EV_KEY , KEY_OK                  },
+	{ "KEY_SELECT"              , EV_KEY , KEY_SELECT              },
+	{ "KEY_GOTO"                , EV_KEY , KEY_GOTO                },
+	{ "KEY_CLEAR"               , EV_KEY , KEY_CLEAR               },
+	{ "KEY_POWER2"              , EV_KEY , KEY_POWER2              },
+	{ "KEY_OPTION"              , EV_KEY , KEY_OPTION              },
+	{ "KEY_INFO"                , EV_KEY , KEY_INFO                },
+	{ "KEY_TIME"                , EV_KEY , KEY_TIME                },
+	{ "KEY_VENDOR"              , EV_KEY , KEY_VENDOR              },
+	{ "KEY_ARCHIVE"             , EV_KEY , KEY_ARCHIVE             },
+	{ "KEY_PROGRAM"             , EV_KEY , KEY_PROGRAM             },
+	{ "KEY_CHANNEL"             , EV_KEY , KEY_CHANNEL             },
+	{ "KEY_FAVORITES"           , EV_KEY , KEY_FAVORITES           },
+	{ "KEY_EPG"                 , EV_KEY , KEY_EPG                 },
+	{ "KEY_PVR"                 , EV_KEY , KEY_PVR                 },
+	{ "KEY_MHP"                 , EV_KEY , KEY_MHP                 },
+	{ "KEY_LANGUAGE"            , EV_KEY , KEY_LANGUAGE            },
+	{ "KEY_TITLE"               , EV_KEY , KEY_TITLE               },
+	{ "KEY_SUBTITLE"            , EV_KEY , KEY_SUBTITLE            },
+	{ "KEY_ANGLE"               , EV_KEY , KEY_ANGLE               },
+	{ "KEY_ZOOM"                , EV_KEY , KEY_ZOOM                },
+	{ "KEY_MODE"                , EV_KEY , KEY_MODE                },
+	{ "KEY_KEYBOARD"            , EV_KEY , KEY_KEYBOARD            },
+	{ "KEY_SCREEN"              , EV_KEY , KEY_SCREEN              },
+	{ "KEY_PC"                  , EV_KEY , KEY_PC                  },
+	{ "KEY_TV"                  , EV_KEY , KEY_TV                  },
+	{ "KEY_TV2"                 , EV_KEY , KEY_TV2                 },
+	{ "KEY_VCR"                 , EV_KEY , KEY_VCR                 },
+	{ "KEY_VCR2"                , EV_KEY , KEY_VCR2                },
+	{ "KEY_SAT"                 , EV_KEY , KEY_SAT                 },
+	{ "KEY_SAT2"                , EV_KEY , KEY_SAT2                },
+	{ "KEY_CD"                  , EV_KEY , KEY_CD                  },
+	{ "KEY_TAPE"                , EV_KEY , KEY_TAPE                },
+	{ "KEY_RADIO"               , EV_KEY , KEY_RADIO               },
+	{ "KEY_TUNER"               , EV_KEY , KEY_TUNER               },
+	{ "KEY_PLAYER"              , EV_KEY , KEY_PLAYER              },
+	{ "KEY_TEXT"                , EV_KEY , KEY_TEXT                },
+	{ "KEY_DVD"                 , EV_KEY , KEY_DVD                 },
+	{ "KEY_AUX"                 , EV_KEY , KEY_AUX                 },
+	{ "KEY_MP3"                 , EV_KEY , KEY_MP3                 },
+	{ "KEY_AUDIO"               , EV_KEY , KEY_AUDIO               },
+	{ "KEY_VIDEO"               , EV_KEY , KEY_VIDEO               },
+	{ "KEY_DIRECTORY"           , EV_KEY , KEY_DIRECTORY           },
+	{ "KEY_LIST"                , EV_KEY , KEY_LIST                },
+	{ "KEY_MEMO"                , EV_KEY , KEY_MEMO                },
+	{ "KEY_CALENDAR"            , EV_KEY , KEY_CALENDAR            },
+	{ "KEY_RED"                 , EV_KEY , KEY_RED                 },
+	{ "KEY_GREEN"               , EV_KEY , KEY_GREEN               },
+	{ "KEY_YELLOW"              , EV_KEY , KEY_YELLOW              },
+	{ "KEY_BLUE"                , EV_KEY , KEY_BLUE                },
+	{ "KEY_CHANNELUP"           , EV_KEY , KEY_CHANNELUP           },
+	{ "KEY_CHANNELDOWN"         , EV_KEY , KEY_CHANNELDOWN         },
+	{ "KEY_FIRST"               , EV_KEY , KEY_FIRST               },
+	{ "KEY_LAST"                , EV_KEY , KEY_LAST                },
+	{ "KEY_AB"                  , EV_KEY , KEY_AB                  },
+	{ "KEY_NEXT"                , EV_KEY , KEY_NEXT                },
+	{ "KEY_RESTART"             , EV_KEY , KEY_RESTART             },
+	{ "KEY_SLOW"                , EV_KEY , KEY_SLOW                },
+	{ "KEY_SHUFFLE"             , EV_KEY , KEY_SHUFFLE             },
+	{ "KEY_BREAK"               , EV_KEY , KEY_BREAK               },
+	{ "KEY_PREVIOUS"            , EV_KEY , KEY_PREVIOUS            },
+	{ "KEY_DIGITS"              , EV_KEY , KEY_DIGITS              },
+	{ "KEY_TEEN"                , EV_KEY , KEY_TEEN                },
+	{ "KEY_TWEN"                , EV_KEY , KEY_TWEN                },
+	{ "KEY_VIDEOPHONE"          , EV_KEY , KEY_VIDEOPHONE          },
+	{ "KEY_GAMES"               , EV_KEY , KEY_GAMES               },
+	{ "KEY_ZOOMIN"              , EV_KEY , KEY_ZOOMIN              },
+	{ "KEY_ZOOMOUT"             , EV_KEY , KEY_ZOOMOUT             },
+	{ "KEY_ZOOMRESET"           , EV_KEY , KEY_ZOOMRESET           },
+	{ "KEY_WORDPROCESSOR"       , EV_KEY , KEY_WORDPROCESSOR       },
+	{ "KEY_EDITOR"              , EV_KEY , KEY_EDITOR              },
+	{ "KEY_SPREADSHEET"         , EV_KEY , KEY_SPREADSHEET         },
+	{ "KEY_GRAPHICSEDITOR"      , EV_KEY , KEY_GRAPHICSEDITOR      },
+	{ "KEY_PRESENTATION"        , EV_KEY , KEY_PRESENTATION        },
+	{ "KEY_DATABASE"            , EV_KEY , KEY_DATABASE            },
+	{ "KEY_NEWS"                , EV_KEY , KEY_NEWS                },
+	{ "KEY_VOICEMAIL"           , EV_KEY , KEY_VOICEMAIL           },
+	{ "KEY_ADDRESSBOOK"         , EV_KEY , KEY_ADDRESSBOOK         },
+	{ "KEY_MESSENGER"           , EV_KEY , KEY_MESSENGER           },
+	{ "KEY_DISPLAYTOGGLE"       , EV_KEY , KEY_DISPLAYTOGGLE       },
+	{ "KEY_SPELLCHECK"          , EV_KEY , KEY_SPELLCHECK          },
+	{ "KEY_LOGOFF"              , EV_KEY , KEY_LOGOFF              },
+	{ "KEY_DOLLAR"              , EV_KEY , KEY_DOLLAR              },
+	{ "KEY_EURO"                , EV_KEY , KEY_EURO                },
+	{ "KEY_FRAMEBACK"           , EV_KEY , KEY_FRAMEBACK           },
+	{ "KEY_FRAMEFORWARD"        , EV_KEY , KEY_FRAMEFORWARD        },
+	{ "KEY_CONTEXT_MENU"        , EV_KEY , KEY_CONTEXT_MENU        },
+	{ "KEY_MEDIA_REPEAT"        , EV_KEY , KEY_MEDIA_REPEAT        },
+	{ "KEY_10CHANNELSUP"        , EV_KEY , KEY_10CHANNELSUP        },
+	{ "KEY_10CHANNELSDOWN"      , EV_KEY , KEY_10CHANNELSDOWN      },
+	{ "KEY_DEL_EOL"             , EV_KEY , KEY_DEL_EOL             },
+	{ "KEY_DEL_EOS"             , EV_KEY , KEY_DEL_EOS             },
+	{ "KEY_INS_LINE"            , EV_KEY , KEY_INS_LINE            },
+	{ "KEY_DEL_LINE"            , EV_KEY , KEY_DEL_LINE            },
+	{ "KEY_FN"                  , EV_KEY , KEY_FN                  },
+	{ "KEY_FN_ESC"              , EV_KEY , KEY_FN_ESC              },
+	{ "KEY_FN_F1"               , EV_KEY , KEY_FN_F1               },
+	{ "KEY_FN_F2"               , EV_KEY , KEY_FN_F2               },
+	{ "KEY_FN_F3"               , EV_KEY , KEY_FN_F3               },
+	{ "KEY_FN_F4"               , EV_KEY , KEY_FN_F4               },
+	{ "KEY_FN_F5"               , EV_KEY , KEY_FN_F5               },
+	{ "KEY_FN_F6"               , EV_KEY , KEY_FN_F6               },
+	{ "KEY_FN_F7"               , EV_KEY , KEY_FN_F7               },
+	{ "KEY_FN_F8"               , EV_KEY , KEY_FN_F8               },
+	{ "KEY_FN_F9"               , EV_KEY , KEY_FN_F9               },
+	{ "KEY_FN_F10"              , EV_KEY , KEY_FN_F10              },
+	{ "KEY_FN_F11"              , EV_KEY , KEY_FN_F11              },
+	{ "KEY_FN_F12"              , EV_KEY , KEY_FN_F12              },
+	{ "KEY_FN_1"                , EV_KEY , KEY_FN_1                },
+	{ "KEY_FN_2"                , EV_KEY , KEY_FN_2                },
+	{ "KEY_FN_D"                , EV_KEY , KEY_FN_D                },
+	{ "KEY_FN_E"                , EV_KEY , KEY_FN_E                },
+	{ "KEY_FN_F"                , EV_KEY , KEY_FN_F                },
+	{ "KEY_FN_S"                , EV_KEY , KEY_FN_S                },
+	{ "KEY_FN_B"                , EV_KEY , KEY_FN_B                },
+	{ "KEY_BRL_DOT1"            , EV_KEY , KEY_BRL_DOT1            },
+	{ "KEY_BRL_DOT2"            , EV_KEY , KEY_BRL_DOT2            },
+	{ "KEY_BRL_DOT3"            , EV_KEY , KEY_BRL_DOT3            },
+	{ "KEY_BRL_DOT4"            , EV_KEY , KEY_BRL_DOT4            },
+	{ "KEY_BRL_DOT5"            , EV_KEY , KEY_BRL_DOT5            },
+	{ "KEY_BRL_DOT6"            , EV_KEY , KEY_BRL_DOT6            },
+	{ "KEY_BRL_DOT7"            , EV_KEY , KEY_BRL_DOT7            },
+	{ "KEY_BRL_DOT8"            , EV_KEY , KEY_BRL_DOT8            },
+	{ "KEY_BRL_DOT9"            , EV_KEY , KEY_BRL_DOT9            },
+	{ "KEY_BRL_DOT10"           , EV_KEY , KEY_BRL_DOT10           },
+	{ "KEY_NUMERIC_0"           , EV_KEY , KEY_NUMERIC_0           },
+	{ "KEY_NUMERIC_1"           , EV_KEY , KEY_NUMERIC_1           },
+	{ "KEY_NUMERIC_2"           , EV_KEY , KEY_NUMERIC_2           },
+	{ "KEY_NUMERIC_3"           , EV_KEY , KEY_NUMERIC_3           },
+	{ "KEY_NUMERIC_4"           , EV_KEY , KEY_NUMERIC_4           },
+	{ "KEY_NUMERIC_5"           , EV_KEY , KEY_NUMERIC_5           },
+	{ "KEY_NUMERIC_6"           , EV_KEY , KEY_NUMERIC_6           },
+	{ "KEY_NUMERIC_7"           , EV_KEY , KEY_NUMERIC_7           },
+	{ "KEY_NUMERIC_8"           , EV_KEY , KEY_NUMERIC_8           },
+	{ "KEY_NUMERIC_9"           , EV_KEY , KEY_NUMERIC_9           },
+	{ "KEY_NUMERIC_STAR"        , EV_KEY , KEY_NUMERIC_STAR        },
+	{ "KEY_NUMERIC_POUND"       , EV_KEY , KEY_NUMERIC_POUND       },
+	{ "KEY_CAMERA_FOCUS"        , EV_KEY , KEY_CAMERA_FOCUS        },
+	{ "KEY_WPS_BUTTON"          , EV_KEY , KEY_WPS_BUTTON          },
+	{ "KEY_TOUCHPAD_TOGGLE"     , EV_KEY , KEY_TOUCHPAD_TOGGLE     },
+	{ "KEY_TOUCHPAD_ON"         , EV_KEY , KEY_TOUCHPAD_ON         },
+	{ "KEY_TOUCHPAD_OFF"        , EV_KEY , KEY_TOUCHPAD_OFF        },
+	{ "BTN_TRIGGER_HAPPY"       , EV_KEY , BTN_TRIGGER_HAPPY       },
+	{ "BTN_TRIGGER_HAPPY1"      , EV_KEY , BTN_TRIGGER_HAPPY1      },
+	{ "BTN_TRIGGER_HAPPY2"      , EV_KEY , BTN_TRIGGER_HAPPY2      },
+	{ "BTN_TRIGGER_HAPPY3"      , EV_KEY , BTN_TRIGGER_HAPPY3      },
+	{ "BTN_TRIGGER_HAPPY4"      , EV_KEY , BTN_TRIGGER_HAPPY4      },
+	{ "BTN_TRIGGER_HAPPY5"      , EV_KEY , BTN_TRIGGER_HAPPY5      },
+	{ "BTN_TRIGGER_HAPPY6"      , EV_KEY , BTN_TRIGGER_HAPPY6      },
+	{ "BTN_TRIGGER_HAPPY7"      , EV_KEY , BTN_TRIGGER_HAPPY7      },
+	{ "BTN_TRIGGER_HAPPY8"      , EV_KEY , BTN_TRIGGER_HAPPY8      },
+	{ "BTN_TRIGGER_HAPPY9"      , EV_KEY , BTN_TRIGGER_HAPPY9      },
+	{ "BTN_TRIGGER_HAPPY10"     , EV_KEY , BTN_TRIGGER_HAPPY10     },
+	{ "BTN_TRIGGER_HAPPY11"     , EV_KEY , BTN_TRIGGER_HAPPY11     },
+	{ "BTN_TRIGGER_HAPPY12"     , EV_KEY , BTN_TRIGGER_HAPPY12     },
+	{ "BTN_TRIGGER_HAPPY13"     , EV_KEY , BTN_TRIGGER_HAPPY13     },
+	{ "BTN_TRIGGER_HAPPY14"     , EV_KEY , BTN_TRIGGER_HAPPY14     },
+	{ "BTN_TRIGGER_HAPPY15"     , EV_KEY , BTN_TRIGGER_HAPPY15     },
+	{ "BTN_TRIGGER_HAPPY16"     , EV_KEY , BTN_TRIGGER_HAPPY16     },
+	{ "BTN_TRIGGER_HAPPY17"     , EV_KEY , BTN_TRIGGER_HAPPY17     },
+	{ "BTN_TRIGGER_HAPPY18"     , EV_KEY , BTN_TRIGGER_HAPPY18     },
+	{ "BTN_TRIGGER_HAPPY19"     , EV_KEY , BTN_TRIGGER_HAPPY19     },
+	{ "BTN_TRIGGER_HAPPY20"     , EV_KEY , BTN_TRIGGER_HAPPY20     },
+	{ "BTN_TRIGGER_HAPPY21"     , EV_KEY , BTN_TRIGGER_HAPPY21     },
+	{ "BTN_TRIGGER_HAPPY22"     , EV_KEY , BTN_TRIGGER_HAPPY22     },
+	{ "BTN_TRIGGER_HAPPY23"     , EV_KEY , BTN_TRIGGER_HAPPY23     },
+	{ "BTN_TRIGGER_HAPPY24"     , EV_KEY , BTN_TRIGGER_HAPPY24     },
+	{ "BTN_TRIGGER_HAPPY25"     , EV_KEY , BTN_TRIGGER_HAPPY25     },
+	{ "BTN_TRIGGER_HAPPY26"     , EV_KEY , BTN_TRIGGER_HAPPY26     },
+	{ "BTN_TRIGGER_HAPPY27"     , EV_KEY , BTN_TRIGGER_HAPPY27     },
+	{ "BTN_TRIGGER_HAPPY28"     , EV_KEY , BTN_TRIGGER_HAPPY28     },
+	{ "BTN_TRIGGER_HAPPY29"     , EV_KEY , BTN_TRIGGER_HAPPY29     },
+	{ "BTN_TRIGGER_HAPPY30"     , EV_KEY , BTN_TRIGGER_HAPPY30     },
+	{ "BTN_TRIGGER_HAPPY31"     , EV_KEY , BTN_TRIGGER_HAPPY31     },
+	{ "BTN_TRIGGER_HAPPY32"     , EV_KEY , BTN_TRIGGER_HAPPY32     },
+	{ "BTN_TRIGGER_HAPPY33"     , EV_KEY , BTN_TRIGGER_HAPPY33     },
+	{ "BTN_TRIGGER_HAPPY34"     , EV_KEY , BTN_TRIGGER_HAPPY34     },
+	{ "BTN_TRIGGER_HAPPY35"     , EV_KEY , BTN_TRIGGER_HAPPY35     },
+	{ "BTN_TRIGGER_HAPPY36"     , EV_KEY , BTN_TRIGGER_HAPPY36     },
+	{ "BTN_TRIGGER_HAPPY37"     , EV_KEY , BTN_TRIGGER_HAPPY37     },
+	{ "BTN_TRIGGER_HAPPY38"     , EV_KEY , BTN_TRIGGER_HAPPY38     },
+	{ "BTN_TRIGGER_HAPPY39"     , EV_KEY , BTN_TRIGGER_HAPPY39     },
+	{ "BTN_TRIGGER_HAPPY40"     , EV_KEY , BTN_TRIGGER_HAPPY40     },
+	{ "REL_X"                   , EV_REL , REL_X                   },
+	{ "REL_Y"                   , EV_REL , REL_Y                   },
+	{ "REL_Z"                   , EV_REL , REL_Z                   },
+	{ "REL_RX"                  , EV_REL , REL_RX                  },
+	{ "REL_RY"                  , EV_REL , REL_RY                  },
+	{ "REL_RZ"                  , EV_REL , REL_RZ                  },
+	{ "REL_HWHEEL"              , EV_REL , REL_HWHEEL              },
+	{ "REL_DIAL"                , EV_REL , REL_DIAL                },
+	{ "REL_WHEEL"               , EV_REL , REL_WHEEL               },
+	{ "REL_MISC"                , EV_REL , REL_MISC                },
+	{ "ABS_X"                   , EV_ABS , ABS_X                   },
+	{ "ABS_Y"                   , EV_ABS , ABS_Y                   },
+	{ "ABS_Z"                   , EV_ABS , ABS_Z                   },
+	{ "ABS_RX"                  , EV_ABS , ABS_RX                  },
+	{ "ABS_RY"                  , EV_ABS , ABS_RY                  },
+	{ "ABS_RZ"                  , EV_ABS , ABS_RZ                  },
+	{ "ABS_THROTTLE"            , EV_ABS , ABS_THROTTLE            },
+	{ "ABS_RUDDER"              , EV_ABS , ABS_RUDDER              },
+	{ "ABS_WHEEL"               , EV_ABS , ABS_WHEEL               },
+	{ "ABS_GAS"                 , EV_ABS , ABS_GAS                 },
+	{ "ABS_BRAKE"               , EV_ABS , ABS_BRAKE               },
+	{ "ABS_HAT0X"               , EV_ABS , ABS_HAT0X               },
+	{ "ABS_HAT0Y"               , EV_ABS , ABS_HAT0Y               },
+	{ "ABS_HAT1X"               , EV_ABS , ABS_HAT1X               },
+	{ "ABS_HAT1Y"               , EV_ABS , ABS_HAT1Y               },
+	{ "ABS_HAT2X"               , EV_ABS , ABS_HAT2X               },
+	{ "ABS_HAT2Y"               , EV_ABS , ABS_HAT2Y               },
+	{ "ABS_HAT3X"               , EV_ABS , ABS_HAT3X               },
+	{ "ABS_HAT3Y"               , EV_ABS , ABS_HAT3Y               },
+	{ "ABS_PRESSURE"            , EV_ABS , ABS_PRESSURE            },
+	{ "ABS_DISTANCE"            , EV_ABS , ABS_DISTANCE            },
+	{ "ABS_TILT_X"              , EV_ABS , ABS_TILT_X              },
+	{ "ABS_TILT_Y"              , EV_ABS , ABS_TILT_Y              },
+	{ "ABS_TOOL_WIDTH"          , EV_ABS , ABS_TOOL_WIDTH          },
+	{ "ABS_VOLUME"              , EV_ABS , ABS_VOLUME              },
+	{ "ABS_MISC"                , EV_ABS , ABS_MISC                },
+	{ "ABS_MT_SLOT"             , EV_ABS , ABS_MT_SLOT             },
+	{ "ABS_MT_TOUCH_MAJOR"      , EV_ABS , ABS_MT_TOUCH_MAJOR      },
+	{ "ABS_MT_TOUCH_MINOR"      , EV_ABS , ABS_MT_TOUCH_MINOR      },
+	{ "ABS_MT_WIDTH_MAJOR"      , EV_ABS , ABS_MT_WIDTH_MAJOR      },
+	{ "ABS_MT_WIDTH_MINOR"      , EV_ABS , ABS_MT_WIDTH_MINOR      },
+	{ "ABS_MT_ORIENTATION"      , EV_ABS , ABS_MT_ORIENTATION      },
+	{ "ABS_MT_POSITION_X"       , EV_ABS , ABS_MT_POSITION_X       },
+	{ "ABS_MT_POSITION_Y"       , EV_ABS , ABS_MT_POSITION_Y       },
+	{ "ABS_MT_TOOL_TYPE"        , EV_ABS , ABS_MT_TOOL_TYPE        },
+	{ "ABS_MT_BLOB_ID"          , EV_ABS , ABS_MT_BLOB_ID          },
+	{ "ABS_MT_TRACKING_ID"      , EV_ABS , ABS_MT_TRACKING_ID      },
+	{ "ABS_MT_PRESSURE"         , EV_ABS , ABS_MT_PRESSURE         },
+	{ "ABS_MT_DISTANCE"         , EV_ABS , ABS_MT_DISTANCE         },
+	{ "SW_LID"                  , EV_SW  , SW_LID                  },
+	{ "SW_TABLET_MODE"          , EV_SW  , SW_TABLET_MODE          },
+	{ "SW_HEADPHONE_INSERT"     , EV_SW  , SW_HEADPHONE_INSERT     },
+	{ "SW_RFKILL_ALL"           , EV_SW  , SW_RFKILL_ALL           },
+	{ "SW_RADIO"                , EV_SW  , SW_RADIO                },
+	{ "SW_MICROPHONE_INSERT"    , EV_SW  , SW_MICROPHONE_INSERT    },
+	{ "SW_DOCK"                 , EV_SW  , SW_DOCK                 },
+	{ "SW_LINEOUT_INSERT"       , EV_SW  , SW_LINEOUT_INSERT       },
+	{ "SW_JACK_PHYSICAL_INSERT" , EV_SW  , SW_JACK_PHYSICAL_INSERT },
+	{ "SW_VIDEOOUT_INSERT"      , EV_SW  , SW_VIDEOOUT_INSERT      },
+	{ "SW_CAMERA_LENS_COVER"    , EV_SW  , SW_CAMERA_LENS_COVER    },
+	{ "SW_KEYPAD_SLIDE"         , EV_SW  , SW_KEYPAD_SLIDE         },
+	{ "SW_FRONT_PROXIMITY"      , EV_SW  , SW_FRONT_PROXIMITY      },
+	{ "SW_ROTATE_LOCK"          , EV_SW  , SW_ROTATE_LOCK          },
+	{ "MSC_SERIAL"              , EV_MSC , MSC_SERIAL              },
+	{ "MSC_PULSELED"            , EV_MSC , MSC_PULSELED            },
+	{ "MSC_GESTURE"             , EV_MSC , MSC_GESTURE             },
+	{ "MSC_RAW"                 , EV_MSC , MSC_RAW                 },
+	{ "MSC_SCAN"                , EV_MSC , MSC_SCAN                },
+	{ "LED_NUML"                , EV_LED , LED_NUML                },
+	{ "LED_CAPSL"               , EV_LED , LED_CAPSL               },
+	{ "LED_SCROLLL"             , EV_LED , LED_SCROLLL             },
+	{ "LED_COMPOSE"             , EV_LED , LED_COMPOSE             },
+	{ "LED_KANA"                , EV_LED , LED_KANA                },
+	{ "LED_SLEEP"               , EV_LED , LED_SLEEP               },
+	{ "LED_SUSPEND"             , EV_LED , LED_SUSPEND             },
+	{ "LED_MUTE"                , EV_LED , LED_MUTE                },
+	{ "LED_MISC"                , EV_LED , LED_MISC                },
+	{ "LED_MAIL"                , EV_LED , LED_MAIL                },
+	{ "LED_CHARGING"            , EV_LED , LED_CHARGING            },
+	{ "SND_CLICK"               , EV_SND , SND_CLICK               },
+	{ "SND_BELL"                , EV_SND , SND_BELL                },
+	{ "SND_TONE"                , EV_SND , SND_TONE                },
+	{ NULL                      , 0      , 0                       },
+};
+#endif
--- /dev/null
+++ eventlircd-0.0.1+svn20110409.0930/src/evkey_code_to_name.h
@@ -0,0 +1,491 @@
+#ifndef _EVENTLIRCD_EVKEY_CODE_TO_NAME_H_
+#define _EVENTLIRCD_EVKEY_CODE_TO_NAME_H_ 1
+#include <///usr/include/linux/input.h>
+static const char *evkey_code_to_name[KEY_MAX] = {
+	[ KEY_ESC              ] = "KEY_ESC",
+	[ KEY_1                ] = "KEY_1",
+	[ KEY_2                ] = "KEY_2",
+	[ KEY_3                ] = "KEY_3",
+	[ KEY_4                ] = "KEY_4",
+	[ KEY_5                ] = "KEY_5",
+	[ KEY_6                ] = "KEY_6",
+	[ KEY_7                ] = "KEY_7",
+	[ KEY_8                ] = "KEY_8",
+	[ KEY_9                ] = "KEY_9",
+	[ KEY_0                ] = "KEY_0",
+	[ KEY_MINUS            ] = "KEY_MINUS",
+	[ KEY_EQUAL            ] = "KEY_EQUAL",
+	[ KEY_BACKSPACE        ] = "KEY_BACKSPACE",
+	[ KEY_TAB              ] = "KEY_TAB",
+	[ KEY_Q                ] = "KEY_Q",
+	[ KEY_W                ] = "KEY_W",
+	[ KEY_E                ] = "KEY_E",
+	[ KEY_R                ] = "KEY_R",
+	[ KEY_T                ] = "KEY_T",
+	[ KEY_Y                ] = "KEY_Y",
+	[ KEY_U                ] = "KEY_U",
+	[ KEY_I                ] = "KEY_I",
+	[ KEY_O                ] = "KEY_O",
+	[ KEY_P                ] = "KEY_P",
+	[ KEY_LEFTBRACE        ] = "KEY_LEFTBRACE",
+	[ KEY_RIGHTBRACE       ] = "KEY_RIGHTBRACE",
+	[ KEY_ENTER            ] = "KEY_ENTER",
+	[ KEY_LEFTCTRL         ] = "KEY_LEFTCTRL",
+	[ KEY_A                ] = "KEY_A",
+	[ KEY_S                ] = "KEY_S",
+	[ KEY_D                ] = "KEY_D",
+	[ KEY_F                ] = "KEY_F",
+	[ KEY_G                ] = "KEY_G",
+	[ KEY_H                ] = "KEY_H",
+	[ KEY_J                ] = "KEY_J",
+	[ KEY_K                ] = "KEY_K",
+	[ KEY_L                ] = "KEY_L",
+	[ KEY_SEMICOLON        ] = "KEY_SEMICOLON",
+	[ KEY_APOSTROPHE       ] = "KEY_APOSTROPHE",
+	[ KEY_GRAVE            ] = "KEY_GRAVE",
+	[ KEY_LEFTSHIFT        ] = "KEY_LEFTSHIFT",
+	[ KEY_BACKSLASH        ] = "KEY_BACKSLASH",
+	[ KEY_Z                ] = "KEY_Z",
+	[ KEY_X                ] = "KEY_X",
+	[ KEY_C                ] = "KEY_C",
+	[ KEY_V                ] = "KEY_V",
+	[ KEY_B                ] = "KEY_B",
+	[ KEY_N                ] = "KEY_N",
+	[ KEY_M                ] = "KEY_M",
+	[ KEY_COMMA            ] = "KEY_COMMA",
+	[ KEY_DOT              ] = "KEY_DOT",
+	[ KEY_SLASH            ] = "KEY_SLASH",
+	[ KEY_RIGHTSHIFT       ] = "KEY_RIGHTSHIFT",
+	[ KEY_KPASTERISK       ] = "KEY_KPASTERISK",
+	[ KEY_LEFTALT          ] = "KEY_LEFTALT",
+	[ KEY_SPACE            ] = "KEY_SPACE",
+	[ KEY_CAPSLOCK         ] = "KEY_CAPSLOCK",
+	[ KEY_F1               ] = "KEY_F1",
+	[ KEY_F2               ] = "KEY_F2",
+	[ KEY_F3               ] = "KEY_F3",
+	[ KEY_F4               ] = "KEY_F4",
+	[ KEY_F5               ] = "KEY_F5",
+	[ KEY_F6               ] = "KEY_F6",
+	[ KEY_F7               ] = "KEY_F7",
+	[ KEY_F8               ] = "KEY_F8",
+	[ KEY_F9               ] = "KEY_F9",
+	[ KEY_F10              ] = "KEY_F10",
+	[ KEY_NUMLOCK          ] = "KEY_NUMLOCK",
+	[ KEY_SCROLLLOCK       ] = "KEY_SCROLLLOCK",
+	[ KEY_KP7              ] = "KEY_KP7",
+	[ KEY_KP8              ] = "KEY_KP8",
+	[ KEY_KP9              ] = "KEY_KP9",
+	[ KEY_KPMINUS          ] = "KEY_KPMINUS",
+	[ KEY_KP4              ] = "KEY_KP4",
+	[ KEY_KP5              ] = "KEY_KP5",
+	[ KEY_KP6              ] = "KEY_KP6",
+	[ KEY_KPPLUS           ] = "KEY_KPPLUS",
+	[ KEY_KP1              ] = "KEY_KP1",
+	[ KEY_KP2              ] = "KEY_KP2",
+	[ KEY_KP3              ] = "KEY_KP3",
+	[ KEY_KP0              ] = "KEY_KP0",
+	[ KEY_KPDOT            ] = "KEY_KPDOT",
+	[ KEY_ZENKAKUHANKAKU   ] = "KEY_ZENKAKUHANKAKU",
+	[ KEY_102ND            ] = "KEY_102ND",
+	[ KEY_F11              ] = "KEY_F11",
+	[ KEY_F12              ] = "KEY_F12",
+	[ KEY_RO               ] = "KEY_RO",
+	[ KEY_KATAKANA         ] = "KEY_KATAKANA",
+	[ KEY_HIRAGANA         ] = "KEY_HIRAGANA",
+	[ KEY_HENKAN           ] = "KEY_HENKAN",
+	[ KEY_KATAKANAHIRAGANA ] = "KEY_KATAKANAHIRAGANA",
+	[ KEY_MUHENKAN         ] = "KEY_MUHENKAN",
+	[ KEY_KPJPCOMMA        ] = "KEY_KPJPCOMMA",
+	[ KEY_KPENTER          ] = "KEY_KPENTER",
+	[ KEY_RIGHTCTRL        ] = "KEY_RIGHTCTRL",
+	[ KEY_KPSLASH          ] = "KEY_KPSLASH",
+	[ KEY_SYSRQ            ] = "KEY_SYSRQ",
+	[ KEY_RIGHTALT         ] = "KEY_RIGHTALT",
+	[ KEY_LINEFEED         ] = "KEY_LINEFEED",
+	[ KEY_HOME             ] = "KEY_HOME",
+	[ KEY_UP               ] = "KEY_UP",
+	[ KEY_PAGEUP           ] = "KEY_PAGEUP",
+	[ KEY_LEFT             ] = "KEY_LEFT",
+	[ KEY_RIGHT            ] = "KEY_RIGHT",
+	[ KEY_END              ] = "KEY_END",
+	[ KEY_DOWN             ] = "KEY_DOWN",
+	[ KEY_PAGEDOWN         ] = "KEY_PAGEDOWN",
+	[ KEY_INSERT           ] = "KEY_INSERT",
+	[ KEY_DELETE           ] = "KEY_DELETE",
+	[ KEY_MACRO            ] = "KEY_MACRO",
+	[ KEY_MUTE             ] = "KEY_MUTE",
+	[ KEY_VOLUMEDOWN       ] = "KEY_VOLUMEDOWN",
+	[ KEY_VOLUMEUP         ] = "KEY_VOLUMEUP",
+	[ KEY_POWER            ] = "KEY_POWER",
+	[ KEY_KPEQUAL          ] = "KEY_KPEQUAL",
+	[ KEY_KPPLUSMINUS      ] = "KEY_KPPLUSMINUS",
+	[ KEY_PAUSE            ] = "KEY_PAUSE",
+	[ KEY_SCALE            ] = "KEY_SCALE",
+	[ KEY_KPCOMMA          ] = "KEY_KPCOMMA",
+	[ KEY_HANGEUL          ] = "KEY_HANGEUL",
+	[ KEY_HANJA            ] = "KEY_HANJA",
+	[ KEY_YEN              ] = "KEY_YEN",
+	[ KEY_LEFTMETA         ] = "KEY_LEFTMETA",
+	[ KEY_RIGHTMETA        ] = "KEY_RIGHTMETA",
+	[ KEY_COMPOSE          ] = "KEY_COMPOSE",
+	[ KEY_STOP             ] = "KEY_STOP",
+	[ KEY_AGAIN            ] = "KEY_AGAIN",
+	[ KEY_PROPS            ] = "KEY_PROPS",
+	[ KEY_UNDO             ] = "KEY_UNDO",
+	[ KEY_FRONT            ] = "KEY_FRONT",
+	[ KEY_COPY             ] = "KEY_COPY",
+	[ KEY_OPEN             ] = "KEY_OPEN",
+	[ KEY_PASTE            ] = "KEY_PASTE",
+	[ KEY_FIND             ] = "KEY_FIND",
+	[ KEY_CUT              ] = "KEY_CUT",
+	[ KEY_HELP             ] = "KEY_HELP",
+	[ KEY_MENU             ] = "KEY_MENU",
+	[ KEY_CALC             ] = "KEY_CALC",
+	[ KEY_SETUP            ] = "KEY_SETUP",
+	[ KEY_SLEEP            ] = "KEY_SLEEP",
+	[ KEY_WAKEUP           ] = "KEY_WAKEUP",
+	[ KEY_FILE             ] = "KEY_FILE",
+	[ KEY_SENDFILE         ] = "KEY_SENDFILE",
+	[ KEY_DELETEFILE       ] = "KEY_DELETEFILE",
+	[ KEY_XFER             ] = "KEY_XFER",
+	[ KEY_PROG1            ] = "KEY_PROG1",
+	[ KEY_PROG2            ] = "KEY_PROG2",
+	[ KEY_WWW              ] = "KEY_WWW",
+	[ KEY_MSDOS            ] = "KEY_MSDOS",
+	[ KEY_COFFEE           ] = "KEY_COFFEE",
+	[ KEY_DIRECTION        ] = "KEY_DIRECTION",
+	[ KEY_CYCLEWINDOWS     ] = "KEY_CYCLEWINDOWS",
+	[ KEY_MAIL             ] = "KEY_MAIL",
+	[ KEY_BOOKMARKS        ] = "KEY_BOOKMARKS",
+	[ KEY_COMPUTER         ] = "KEY_COMPUTER",
+	[ KEY_BACK             ] = "KEY_BACK",
+	[ KEY_FORWARD          ] = "KEY_FORWARD",
+	[ KEY_CLOSECD          ] = "KEY_CLOSECD",
+	[ KEY_EJECTCD          ] = "KEY_EJECTCD",
+	[ KEY_EJECTCLOSECD     ] = "KEY_EJECTCLOSECD",
+	[ KEY_NEXTSONG         ] = "KEY_NEXTSONG",
+	[ KEY_PLAYPAUSE        ] = "KEY_PLAYPAUSE",
+	[ KEY_PREVIOUSSONG     ] = "KEY_PREVIOUSSONG",
+	[ KEY_STOPCD           ] = "KEY_STOPCD",
+	[ KEY_RECORD           ] = "KEY_RECORD",
+	[ KEY_REWIND           ] = "KEY_REWIND",
+	[ KEY_PHONE            ] = "KEY_PHONE",
+	[ KEY_ISO              ] = "KEY_ISO",
+	[ KEY_CONFIG           ] = "KEY_CONFIG",
+	[ KEY_HOMEPAGE         ] = "KEY_HOMEPAGE",
+	[ KEY_REFRESH          ] = "KEY_REFRESH",
+	[ KEY_EXIT             ] = "KEY_EXIT",
+	[ KEY_MOVE             ] = "KEY_MOVE",
+	[ KEY_EDIT             ] = "KEY_EDIT",
+	[ KEY_SCROLLUP         ] = "KEY_SCROLLUP",
+	[ KEY_SCROLLDOWN       ] = "KEY_SCROLLDOWN",
+	[ KEY_KPLEFTPAREN      ] = "KEY_KPLEFTPAREN",
+	[ KEY_KPRIGHTPAREN     ] = "KEY_KPRIGHTPAREN",
+	[ KEY_NEW              ] = "KEY_NEW",
+	[ KEY_REDO             ] = "KEY_REDO",
+	[ KEY_F13              ] = "KEY_F13",
+	[ KEY_F14              ] = "KEY_F14",
+	[ KEY_F15              ] = "KEY_F15",
+	[ KEY_F16              ] = "KEY_F16",
+	[ KEY_F17              ] = "KEY_F17",
+	[ KEY_F18              ] = "KEY_F18",
+	[ KEY_F19              ] = "KEY_F19",
+	[ KEY_F20              ] = "KEY_F20",
+	[ KEY_F21              ] = "KEY_F21",
+	[ KEY_F22              ] = "KEY_F22",
+	[ KEY_F23              ] = "KEY_F23",
+	[ KEY_F24              ] = "KEY_F24",
+	[ KEY_PLAYCD           ] = "KEY_PLAYCD",
+	[ KEY_PAUSECD          ] = "KEY_PAUSECD",
+	[ KEY_PROG3            ] = "KEY_PROG3",
+	[ KEY_PROG4            ] = "KEY_PROG4",
+	[ KEY_DASHBOARD        ] = "KEY_DASHBOARD",
+	[ KEY_SUSPEND          ] = "KEY_SUSPEND",
+	[ KEY_CLOSE            ] = "KEY_CLOSE",
+	[ KEY_PLAY             ] = "KEY_PLAY",
+	[ KEY_FASTFORWARD      ] = "KEY_FASTFORWARD",
+	[ KEY_BASSBOOST        ] = "KEY_BASSBOOST",
+	[ KEY_PRINT            ] = "KEY_PRINT",
+	[ KEY_HP               ] = "KEY_HP",
+	[ KEY_CAMERA           ] = "KEY_CAMERA",
+	[ KEY_SOUND            ] = "KEY_SOUND",
+	[ KEY_QUESTION         ] = "KEY_QUESTION",
+	[ KEY_EMAIL            ] = "KEY_EMAIL",
+	[ KEY_CHAT             ] = "KEY_CHAT",
+	[ KEY_SEARCH           ] = "KEY_SEARCH",
+	[ KEY_CONNECT          ] = "KEY_CONNECT",
+	[ KEY_FINANCE          ] = "KEY_FINANCE",
+	[ KEY_SPORT            ] = "KEY_SPORT",
+	[ KEY_SHOP             ] = "KEY_SHOP",
+	[ KEY_ALTERASE         ] = "KEY_ALTERASE",
+	[ KEY_CANCEL           ] = "KEY_CANCEL",
+	[ KEY_BRIGHTNESSDOWN   ] = "KEY_BRIGHTNESSDOWN",
+	[ KEY_BRIGHTNESSUP     ] = "KEY_BRIGHTNESSUP",
+	[ KEY_MEDIA            ] = "KEY_MEDIA",
+	[ KEY_SWITCHVIDEOMODE  ] = "KEY_SWITCHVIDEOMODE",
+	[ KEY_KBDILLUMTOGGLE   ] = "KEY_KBDILLUMTOGGLE",
+	[ KEY_KBDILLUMDOWN     ] = "KEY_KBDILLUMDOWN",
+	[ KEY_KBDILLUMUP       ] = "KEY_KBDILLUMUP",
+	[ KEY_SEND             ] = "KEY_SEND",
+	[ KEY_REPLY            ] = "KEY_REPLY",
+	[ KEY_FORWARDMAIL      ] = "KEY_FORWARDMAIL",
+	[ KEY_SAVE             ] = "KEY_SAVE",
+	[ KEY_DOCUMENTS        ] = "KEY_DOCUMENTS",
+	[ KEY_BATTERY          ] = "KEY_BATTERY",
+	[ KEY_BLUETOOTH        ] = "KEY_BLUETOOTH",
+	[ KEY_WLAN             ] = "KEY_WLAN",
+	[ KEY_UWB              ] = "KEY_UWB",
+	[ KEY_UNKNOWN          ] = "KEY_UNKNOWN",
+	[ KEY_VIDEO_NEXT       ] = "KEY_VIDEO_NEXT",
+	[ KEY_VIDEO_PREV       ] = "KEY_VIDEO_PREV",
+	[ KEY_BRIGHTNESS_CYCLE ] = "KEY_BRIGHTNESS_CYCLE",
+	[ KEY_BRIGHTNESS_ZERO  ] = "KEY_BRIGHTNESS_ZERO",
+	[ KEY_DISPLAY_OFF      ] = "KEY_DISPLAY_OFF",
+	[ KEY_WIMAX            ] = "KEY_WIMAX",
+	[ KEY_RFKILL           ] = "KEY_RFKILL",
+	[ BTN_MISC             ] = "BTN_MISC",
+	[ BTN_1                ] = "BTN_1",
+	[ BTN_2                ] = "BTN_2",
+	[ BTN_3                ] = "BTN_3",
+	[ BTN_4                ] = "BTN_4",
+	[ BTN_5                ] = "BTN_5",
+	[ BTN_6                ] = "BTN_6",
+	[ BTN_7                ] = "BTN_7",
+	[ BTN_8                ] = "BTN_8",
+	[ BTN_9                ] = "BTN_9",
+	[ BTN_MOUSE            ] = "BTN_MOUSE",
+	[ BTN_RIGHT            ] = "BTN_RIGHT",
+	[ BTN_MIDDLE           ] = "BTN_MIDDLE",
+	[ BTN_SIDE             ] = "BTN_SIDE",
+	[ BTN_EXTRA            ] = "BTN_EXTRA",
+	[ BTN_FORWARD          ] = "BTN_FORWARD",
+	[ BTN_BACK             ] = "BTN_BACK",
+	[ BTN_TASK             ] = "BTN_TASK",
+	[ BTN_JOYSTICK         ] = "BTN_JOYSTICK",
+	[ BTN_THUMB            ] = "BTN_THUMB",
+	[ BTN_THUMB2           ] = "BTN_THUMB2",
+	[ BTN_TOP              ] = "BTN_TOP",
+	[ BTN_TOP2             ] = "BTN_TOP2",
+	[ BTN_PINKIE           ] = "BTN_PINKIE",
+	[ BTN_BASE             ] = "BTN_BASE",
+	[ BTN_BASE2            ] = "BTN_BASE2",
+	[ BTN_BASE3            ] = "BTN_BASE3",
+	[ BTN_BASE4            ] = "BTN_BASE4",
+	[ BTN_BASE5            ] = "BTN_BASE5",
+	[ BTN_BASE6            ] = "BTN_BASE6",
+	[ BTN_DEAD             ] = "BTN_DEAD",
+	[ BTN_GAMEPAD          ] = "BTN_GAMEPAD",
+	[ BTN_B                ] = "BTN_B",
+	[ BTN_C                ] = "BTN_C",
+	[ BTN_X                ] = "BTN_X",
+	[ BTN_Y                ] = "BTN_Y",
+	[ BTN_Z                ] = "BTN_Z",
+	[ BTN_TL               ] = "BTN_TL",
+	[ BTN_TR               ] = "BTN_TR",
+	[ BTN_TL2              ] = "BTN_TL2",
+	[ BTN_TR2              ] = "BTN_TR2",
+	[ BTN_SELECT           ] = "BTN_SELECT",
+	[ BTN_START            ] = "BTN_START",
+	[ BTN_MODE             ] = "BTN_MODE",
+	[ BTN_THUMBL           ] = "BTN_THUMBL",
+	[ BTN_THUMBR           ] = "BTN_THUMBR",
+	[ BTN_DIGI             ] = "BTN_DIGI",
+	[ BTN_TOOL_RUBBER      ] = "BTN_TOOL_RUBBER",
+	[ BTN_TOOL_BRUSH       ] = "BTN_TOOL_BRUSH",
+	[ BTN_TOOL_PENCIL      ] = "BTN_TOOL_PENCIL",
+	[ BTN_TOOL_AIRBRUSH    ] = "BTN_TOOL_AIRBRUSH",
+	[ BTN_TOOL_FINGER      ] = "BTN_TOOL_FINGER",
+	[ BTN_TOOL_MOUSE       ] = "BTN_TOOL_MOUSE",
+	[ BTN_TOOL_LENS        ] = "BTN_TOOL_LENS",
+	[ BTN_TOUCH            ] = "BTN_TOUCH",
+	[ BTN_STYLUS           ] = "BTN_STYLUS",
+	[ BTN_STYLUS2          ] = "BTN_STYLUS2",
+	[ BTN_TOOL_DOUBLETAP   ] = "BTN_TOOL_DOUBLETAP",
+	[ BTN_TOOL_TRIPLETAP   ] = "BTN_TOOL_TRIPLETAP",
+	[ BTN_TOOL_QUADTAP     ] = "BTN_TOOL_QUADTAP",
+	[ BTN_WHEEL            ] = "BTN_WHEEL",
+	[ BTN_GEAR_UP          ] = "BTN_GEAR_UP",
+	[ KEY_OK               ] = "KEY_OK",
+	[ KEY_SELECT           ] = "KEY_SELECT",
+	[ KEY_GOTO             ] = "KEY_GOTO",
+	[ KEY_CLEAR            ] = "KEY_CLEAR",
+	[ KEY_POWER2           ] = "KEY_POWER2",
+	[ KEY_OPTION           ] = "KEY_OPTION",
+	[ KEY_INFO             ] = "KEY_INFO",
+	[ KEY_TIME             ] = "KEY_TIME",
+	[ KEY_VENDOR           ] = "KEY_VENDOR",
+	[ KEY_ARCHIVE          ] = "KEY_ARCHIVE",
+	[ KEY_PROGRAM          ] = "KEY_PROGRAM",
+	[ KEY_CHANNEL          ] = "KEY_CHANNEL",
+	[ KEY_FAVORITES        ] = "KEY_FAVORITES",
+	[ KEY_EPG              ] = "KEY_EPG",
+	[ KEY_PVR              ] = "KEY_PVR",
+	[ KEY_MHP              ] = "KEY_MHP",
+	[ KEY_LANGUAGE         ] = "KEY_LANGUAGE",
+	[ KEY_TITLE            ] = "KEY_TITLE",
+	[ KEY_SUBTITLE         ] = "KEY_SUBTITLE",
+	[ KEY_ANGLE            ] = "KEY_ANGLE",
+	[ KEY_ZOOM             ] = "KEY_ZOOM",
+	[ KEY_MODE             ] = "KEY_MODE",
+	[ KEY_KEYBOARD         ] = "KEY_KEYBOARD",
+	[ KEY_SCREEN           ] = "KEY_SCREEN",
+	[ KEY_PC               ] = "KEY_PC",
+	[ KEY_TV               ] = "KEY_TV",
+	[ KEY_TV2              ] = "KEY_TV2",
+	[ KEY_VCR              ] = "KEY_VCR",
+	[ KEY_VCR2             ] = "KEY_VCR2",
+	[ KEY_SAT              ] = "KEY_SAT",
+	[ KEY_SAT2             ] = "KEY_SAT2",
+	[ KEY_CD               ] = "KEY_CD",
+	[ KEY_TAPE             ] = "KEY_TAPE",
+	[ KEY_RADIO            ] = "KEY_RADIO",
+	[ KEY_TUNER            ] = "KEY_TUNER",
+	[ KEY_PLAYER           ] = "KEY_PLAYER",
+	[ KEY_TEXT             ] = "KEY_TEXT",
+	[ KEY_DVD              ] = "KEY_DVD",
+	[ KEY_AUX              ] = "KEY_AUX",
+	[ KEY_MP3              ] = "KEY_MP3",
+	[ KEY_AUDIO            ] = "KEY_AUDIO",
+	[ KEY_VIDEO            ] = "KEY_VIDEO",
+	[ KEY_DIRECTORY        ] = "KEY_DIRECTORY",
+	[ KEY_LIST             ] = "KEY_LIST",
+	[ KEY_MEMO             ] = "KEY_MEMO",
+	[ KEY_CALENDAR         ] = "KEY_CALENDAR",
+	[ KEY_RED              ] = "KEY_RED",
+	[ KEY_GREEN            ] = "KEY_GREEN",
+	[ KEY_YELLOW           ] = "KEY_YELLOW",
+	[ KEY_BLUE             ] = "KEY_BLUE",
+	[ KEY_CHANNELUP        ] = "KEY_CHANNELUP",
+	[ KEY_CHANNELDOWN      ] = "KEY_CHANNELDOWN",
+	[ KEY_FIRST            ] = "KEY_FIRST",
+	[ KEY_LAST             ] = "KEY_LAST",
+	[ KEY_AB               ] = "KEY_AB",
+	[ KEY_NEXT             ] = "KEY_NEXT",
+	[ KEY_RESTART          ] = "KEY_RESTART",
+	[ KEY_SLOW             ] = "KEY_SLOW",
+	[ KEY_SHUFFLE          ] = "KEY_SHUFFLE",
+	[ KEY_BREAK            ] = "KEY_BREAK",
+	[ KEY_PREVIOUS         ] = "KEY_PREVIOUS",
+	[ KEY_DIGITS           ] = "KEY_DIGITS",
+	[ KEY_TEEN             ] = "KEY_TEEN",
+	[ KEY_TWEN             ] = "KEY_TWEN",
+	[ KEY_VIDEOPHONE       ] = "KEY_VIDEOPHONE",
+	[ KEY_GAMES            ] = "KEY_GAMES",
+	[ KEY_ZOOMIN           ] = "KEY_ZOOMIN",
+	[ KEY_ZOOMOUT          ] = "KEY_ZOOMOUT",
+	[ KEY_ZOOMRESET        ] = "KEY_ZOOMRESET",
+	[ KEY_WORDPROCESSOR    ] = "KEY_WORDPROCESSOR",
+	[ KEY_EDITOR           ] = "KEY_EDITOR",
+	[ KEY_SPREADSHEET      ] = "KEY_SPREADSHEET",
+	[ KEY_GRAPHICSEDITOR   ] = "KEY_GRAPHICSEDITOR",
+	[ KEY_PRESENTATION     ] = "KEY_PRESENTATION",
+	[ KEY_DATABASE         ] = "KEY_DATABASE",
+	[ KEY_NEWS             ] = "KEY_NEWS",
+	[ KEY_VOICEMAIL        ] = "KEY_VOICEMAIL",
+	[ KEY_ADDRESSBOOK      ] = "KEY_ADDRESSBOOK",
+	[ KEY_MESSENGER        ] = "KEY_MESSENGER",
+	[ KEY_DISPLAYTOGGLE    ] = "KEY_DISPLAYTOGGLE",
+	[ KEY_SPELLCHECK       ] = "KEY_SPELLCHECK",
+	[ KEY_LOGOFF           ] = "KEY_LOGOFF",
+	[ KEY_DOLLAR           ] = "KEY_DOLLAR",
+	[ KEY_EURO             ] = "KEY_EURO",
+	[ KEY_FRAMEBACK        ] = "KEY_FRAMEBACK",
+	[ KEY_FRAMEFORWARD     ] = "KEY_FRAMEFORWARD",
+	[ KEY_CONTEXT_MENU     ] = "KEY_CONTEXT_MENU",
+	[ KEY_MEDIA_REPEAT     ] = "KEY_MEDIA_REPEAT",
+	[ KEY_10CHANNELSUP     ] = "KEY_10CHANNELSUP",
+	[ KEY_10CHANNELSDOWN   ] = "KEY_10CHANNELSDOWN",
+	[ KEY_DEL_EOL          ] = "KEY_DEL_EOL",
+	[ KEY_DEL_EOS          ] = "KEY_DEL_EOS",
+	[ KEY_INS_LINE         ] = "KEY_INS_LINE",
+	[ KEY_DEL_LINE         ] = "KEY_DEL_LINE",
+	[ KEY_FN               ] = "KEY_FN",
+	[ KEY_FN_ESC           ] = "KEY_FN_ESC",
+	[ KEY_FN_F1            ] = "KEY_FN_F1",
+	[ KEY_FN_F2            ] = "KEY_FN_F2",
+	[ KEY_FN_F3            ] = "KEY_FN_F3",
+	[ KEY_FN_F4            ] = "KEY_FN_F4",
+	[ KEY_FN_F5            ] = "KEY_FN_F5",
+	[ KEY_FN_F6            ] = "KEY_FN_F6",
+	[ KEY_FN_F7            ] = "KEY_FN_F7",
+	[ KEY_FN_F8            ] = "KEY_FN_F8",
+	[ KEY_FN_F9            ] = "KEY_FN_F9",
+	[ KEY_FN_F10           ] = "KEY_FN_F10",
+	[ KEY_FN_F11           ] = "KEY_FN_F11",
+	[ KEY_FN_F12           ] = "KEY_FN_F12",
+	[ KEY_FN_1             ] = "KEY_FN_1",
+	[ KEY_FN_2             ] = "KEY_FN_2",
+	[ KEY_FN_D             ] = "KEY_FN_D",
+	[ KEY_FN_E             ] = "KEY_FN_E",
+	[ KEY_FN_F             ] = "KEY_FN_F",
+	[ KEY_FN_S             ] = "KEY_FN_S",
+	[ KEY_FN_B             ] = "KEY_FN_B",
+	[ KEY_BRL_DOT1         ] = "KEY_BRL_DOT1",
+	[ KEY_BRL_DOT2         ] = "KEY_BRL_DOT2",
+	[ KEY_BRL_DOT3         ] = "KEY_BRL_DOT3",
+	[ KEY_BRL_DOT4         ] = "KEY_BRL_DOT4",
+	[ KEY_BRL_DOT5         ] = "KEY_BRL_DOT5",
+	[ KEY_BRL_DOT6         ] = "KEY_BRL_DOT6",
+	[ KEY_BRL_DOT7         ] = "KEY_BRL_DOT7",
+	[ KEY_BRL_DOT8         ] = "KEY_BRL_DOT8",
+	[ KEY_BRL_DOT9         ] = "KEY_BRL_DOT9",
+	[ KEY_BRL_DOT10        ] = "KEY_BRL_DOT10",
+	[ KEY_NUMERIC_0        ] = "KEY_NUMERIC_0",
+	[ KEY_NUMERIC_1        ] = "KEY_NUMERIC_1",
+	[ KEY_NUMERIC_2        ] = "KEY_NUMERIC_2",
+	[ KEY_NUMERIC_3        ] = "KEY_NUMERIC_3",
+	[ KEY_NUMERIC_4        ] = "KEY_NUMERIC_4",
+	[ KEY_NUMERIC_5        ] = "KEY_NUMERIC_5",
+	[ KEY_NUMERIC_6        ] = "KEY_NUMERIC_6",
+	[ KEY_NUMERIC_7        ] = "KEY_NUMERIC_7",
+	[ KEY_NUMERIC_8        ] = "KEY_NUMERIC_8",
+	[ KEY_NUMERIC_9        ] = "KEY_NUMERIC_9",
+	[ KEY_NUMERIC_STAR     ] = "KEY_NUMERIC_STAR",
+	[ KEY_NUMERIC_POUND    ] = "KEY_NUMERIC_POUND",
+	[ KEY_CAMERA_FOCUS     ] = "KEY_CAMERA_FOCUS",
+	[ KEY_WPS_BUTTON       ] = "KEY_WPS_BUTTON",
+	[ KEY_TOUCHPAD_TOGGLE  ] = "KEY_TOUCHPAD_TOGGLE",
+	[ KEY_TOUCHPAD_ON      ] = "KEY_TOUCHPAD_ON",
+	[ KEY_TOUCHPAD_OFF     ] = "KEY_TOUCHPAD_OFF",
+	[ BTN_TRIGGER_HAPPY    ] = "BTN_TRIGGER_HAPPY",
+	[ BTN_TRIGGER_HAPPY2   ] = "BTN_TRIGGER_HAPPY2",
+	[ BTN_TRIGGER_HAPPY3   ] = "BTN_TRIGGER_HAPPY3",
+	[ BTN_TRIGGER_HAPPY4   ] = "BTN_TRIGGER_HAPPY4",
+	[ BTN_TRIGGER_HAPPY5   ] = "BTN_TRIGGER_HAPPY5",
+	[ BTN_TRIGGER_HAPPY6   ] = "BTN_TRIGGER_HAPPY6",
+	[ BTN_TRIGGER_HAPPY7   ] = "BTN_TRIGGER_HAPPY7",
+	[ BTN_TRIGGER_HAPPY8   ] = "BTN_TRIGGER_HAPPY8",
+	[ BTN_TRIGGER_HAPPY9   ] = "BTN_TRIGGER_HAPPY9",
+	[ BTN_TRIGGER_HAPPY10  ] = "BTN_TRIGGER_HAPPY10",
+	[ BTN_TRIGGER_HAPPY11  ] = "BTN_TRIGGER_HAPPY11",
+	[ BTN_TRIGGER_HAPPY12  ] = "BTN_TRIGGER_HAPPY12",
+	[ BTN_TRIGGER_HAPPY13  ] = "BTN_TRIGGER_HAPPY13",
+	[ BTN_TRIGGER_HAPPY14  ] = "BTN_TRIGGER_HAPPY14",
+	[ BTN_TRIGGER_HAPPY15  ] = "BTN_TRIGGER_HAPPY15",
+	[ BTN_TRIGGER_HAPPY16  ] = "BTN_TRIGGER_HAPPY16",
+	[ BTN_TRIGGER_HAPPY17  ] = "BTN_TRIGGER_HAPPY17",
+	[ BTN_TRIGGER_HAPPY18  ] = "BTN_TRIGGER_HAPPY18",
+	[ BTN_TRIGGER_HAPPY19  ] = "BTN_TRIGGER_HAPPY19",
+	[ BTN_TRIGGER_HAPPY20  ] = "BTN_TRIGGER_HAPPY20",
+	[ BTN_TRIGGER_HAPPY21  ] = "BTN_TRIGGER_HAPPY21",
+	[ BTN_TRIGGER_HAPPY22  ] = "BTN_TRIGGER_HAPPY22",
+	[ BTN_TRIGGER_HAPPY23  ] = "BTN_TRIGGER_HAPPY23",
+	[ BTN_TRIGGER_HAPPY24  ] = "BTN_TRIGGER_HAPPY24",
+	[ BTN_TRIGGER_HAPPY25  ] = "BTN_TRIGGER_HAPPY25",
+	[ BTN_TRIGGER_HAPPY26  ] = "BTN_TRIGGER_HAPPY26",
+	[ BTN_TRIGGER_HAPPY27  ] = "BTN_TRIGGER_HAPPY27",
+	[ BTN_TRIGGER_HAPPY28  ] = "BTN_TRIGGER_HAPPY28",
+	[ BTN_TRIGGER_HAPPY29  ] = "BTN_TRIGGER_HAPPY29",
+	[ BTN_TRIGGER_HAPPY30  ] = "BTN_TRIGGER_HAPPY30",
+	[ BTN_TRIGGER_HAPPY31  ] = "BTN_TRIGGER_HAPPY31",
+	[ BTN_TRIGGER_HAPPY32  ] = "BTN_TRIGGER_HAPPY32",
+	[ BTN_TRIGGER_HAPPY33  ] = "BTN_TRIGGER_HAPPY33",
+	[ BTN_TRIGGER_HAPPY34  ] = "BTN_TRIGGER_HAPPY34",
+	[ BTN_TRIGGER_HAPPY35  ] = "BTN_TRIGGER_HAPPY35",
+	[ BTN_TRIGGER_HAPPY36  ] = "BTN_TRIGGER_HAPPY36",
+	[ BTN_TRIGGER_HAPPY37  ] = "BTN_TRIGGER_HAPPY37",
+	[ BTN_TRIGGER_HAPPY38  ] = "BTN_TRIGGER_HAPPY38",
+	[ BTN_TRIGGER_HAPPY39  ] = "BTN_TRIGGER_HAPPY39",
+	[ BTN_TRIGGER_HAPPY40  ] = "BTN_TRIGGER_HAPPY40",
+};
+#endif
--- /dev/null
+++ eventlircd-0.0.1+svn20110409.0930/src/evkey_type.h
@@ -0,0 +1,494 @@
+#ifndef _EVENTLIRCD_EVKEY_TYPE_H_
+#define _EVENTLIRCD_EVKEY_TYPE_H_ 1
+#include <///usr/include/linux/input.h>
+#define EVENTLIRCD_EVKEY_TYPE_NULL 0
+#define EVENTLIRCD_EVKEY_TYPE_KEY  1
+#define EVENTLIRCD_EVKEY_TYPE_BTN  2
+static const int evkey_type[KEY_MAX] = {
+	[ KEY_ESC              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_1                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_2                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_3                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_4                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_5                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_6                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_7                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_8                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_9                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_0                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MINUS            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EQUAL            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BACKSPACE        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TAB              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_Q                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_W                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_E                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_R                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_T                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_Y                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_U                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_I                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_O                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_P                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LEFTBRACE        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RIGHTBRACE       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ENTER            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LEFTCTRL         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_A                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_S                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_D                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_G                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_H                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_J                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_K                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_L                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SEMICOLON        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_APOSTROPHE       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_GRAVE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LEFTSHIFT        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BACKSLASH        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_Z                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_X                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_C                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_V                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_B                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_N                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_M                ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_COMMA            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DOT              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SLASH            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RIGHTSHIFT       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPASTERISK       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LEFTALT          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SPACE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CAPSLOCK         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F1               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F2               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F3               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F4               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F5               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F6               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F7               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F8               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F9               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F10              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMLOCK          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SCROLLLOCK       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP7              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP8              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP9              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPMINUS          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP4              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP5              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP6              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPPLUS           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP1              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP2              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP3              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KP0              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPDOT            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ZENKAKUHANKAKU   ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_102ND            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F11              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F12              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RO               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KATAKANA         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HIRAGANA         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HENKAN           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KATAKANAHIRAGANA ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MUHENKAN         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPJPCOMMA        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPENTER          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RIGHTCTRL        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPSLASH          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SYSRQ            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RIGHTALT         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LINEFEED         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HOME             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_UP               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PAGEUP           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LEFT             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RIGHT            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_END              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DOWN             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PAGEDOWN         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_INSERT           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DELETE           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MACRO            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MUTE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VOLUMEDOWN       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VOLUMEUP         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_POWER            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPEQUAL          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPPLUSMINUS      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PAUSE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SCALE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPCOMMA          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HANGEUL          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HANJA            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_YEN              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LEFTMETA         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RIGHTMETA        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_COMPOSE          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_STOP             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_AGAIN            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PROPS            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_UNDO             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FRONT            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_COPY             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_OPEN             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PASTE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FIND             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CUT              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HELP             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MENU             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CALC             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SETUP            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SLEEP            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_WAKEUP           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FILE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SENDFILE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DELETEFILE       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_XFER             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PROG1            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PROG2            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_WWW              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MSDOS            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_COFFEE           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DIRECTION        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CYCLEWINDOWS     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MAIL             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BOOKMARKS        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_COMPUTER         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BACK             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FORWARD          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CLOSECD          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EJECTCD          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EJECTCLOSECD     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NEXTSONG         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PLAYPAUSE        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PREVIOUSSONG     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_STOPCD           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RECORD           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_REWIND           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PHONE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ISO              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CONFIG           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HOMEPAGE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_REFRESH          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EXIT             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MOVE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EDIT             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SCROLLUP         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SCROLLDOWN       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPLEFTPAREN      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KPRIGHTPAREN     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NEW              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_REDO             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F13              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F14              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F15              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F16              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F17              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F18              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F19              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F20              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F21              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F22              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F23              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_F24              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PLAYCD           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PAUSECD          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PROG3            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PROG4            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DASHBOARD        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SUSPEND          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CLOSE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PLAY             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FASTFORWARD      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BASSBOOST        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PRINT            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_HP               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CAMERA           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SOUND            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_QUESTION         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EMAIL            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CHAT             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SEARCH           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CONNECT          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FINANCE          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SPORT            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SHOP             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ALTERASE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CANCEL           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRIGHTNESSDOWN   ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRIGHTNESSUP     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MEDIA            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SWITCHVIDEOMODE  ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KBDILLUMTOGGLE   ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KBDILLUMDOWN     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KBDILLUMUP       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SEND             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_REPLY            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FORWARDMAIL      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SAVE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DOCUMENTS        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BATTERY          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BLUETOOTH        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_WLAN             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_UWB              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_UNKNOWN          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VIDEO_NEXT       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VIDEO_PREV       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRIGHTNESS_CYCLE ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRIGHTNESS_ZERO  ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DISPLAY_OFF      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_WIMAX            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RFKILL           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ BTN_MISC             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_1                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_2                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_3                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_4                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_5                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_6                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_7                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_8                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_9                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_MOUSE            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_RIGHT            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_MIDDLE           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_SIDE             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_EXTRA            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_FORWARD          ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BACK             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TASK             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_JOYSTICK         ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_THUMB            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_THUMB2           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOP              ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOP2             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_PINKIE           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BASE             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BASE2            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BASE3            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BASE4            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BASE5            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_BASE6            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_DEAD             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_GAMEPAD          ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_B                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_C                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_X                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_Y                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_Z                ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TL               ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TR               ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TL2              ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TR2              ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_SELECT           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_START            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_MODE             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_THUMBL           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_THUMBR           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_DIGI             ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_RUBBER      ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_BRUSH       ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_PENCIL      ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_AIRBRUSH    ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_FINGER      ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_MOUSE       ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_LENS        ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOUCH            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_STYLUS           ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_STYLUS2          ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_DOUBLETAP   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_TRIPLETAP   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TOOL_QUADTAP     ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_WHEEL            ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_GEAR_UP          ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ KEY_OK               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SELECT           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_GOTO             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CLEAR            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_POWER2           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_OPTION           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_INFO             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TIME             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VENDOR           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ARCHIVE          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PROGRAM          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CHANNEL          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FAVORITES        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EPG              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PVR              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MHP              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LANGUAGE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TITLE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SUBTITLE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ANGLE            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ZOOM             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MODE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_KEYBOARD         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SCREEN           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PC               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TV               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TV2              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VCR              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VCR2             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SAT              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SAT2             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CD               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TAPE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RADIO            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TUNER            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PLAYER           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TEXT             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DVD              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_AUX              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MP3              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_AUDIO            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VIDEO            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DIRECTORY        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LIST             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MEMO             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CALENDAR         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RED              ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_GREEN            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_YELLOW           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BLUE             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CHANNELUP        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CHANNELDOWN      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FIRST            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LAST             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_AB               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NEXT             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_RESTART          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SLOW             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SHUFFLE          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BREAK            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PREVIOUS         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DIGITS           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TEEN             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TWEN             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VIDEOPHONE       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_GAMES            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ZOOMIN           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ZOOMOUT          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ZOOMRESET        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_WORDPROCESSOR    ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EDITOR           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SPREADSHEET      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_GRAPHICSEDITOR   ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_PRESENTATION     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DATABASE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NEWS             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_VOICEMAIL        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_ADDRESSBOOK      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MESSENGER        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DISPLAYTOGGLE    ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_SPELLCHECK       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_LOGOFF           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DOLLAR           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_EURO             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FRAMEBACK        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FRAMEFORWARD     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CONTEXT_MENU     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_MEDIA_REPEAT     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_10CHANNELSUP     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_10CHANNELSDOWN   ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DEL_EOL          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DEL_EOS          ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_INS_LINE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_DEL_LINE         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN               ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_ESC           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F1            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F2            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F3            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F4            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F5            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F6            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F7            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F8            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F9            ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F10           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F11           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F12           ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_1             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_2             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_D             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_E             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_F             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_S             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_FN_B             ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT1         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT2         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT3         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT4         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT5         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT6         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT7         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT8         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT9         ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_BRL_DOT10        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_0        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_1        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_2        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_3        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_4        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_5        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_6        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_7        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_8        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_9        ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_STAR     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_NUMERIC_POUND    ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_CAMERA_FOCUS     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_WPS_BUTTON       ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TOUCHPAD_TOGGLE  ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TOUCHPAD_ON      ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ KEY_TOUCHPAD_OFF     ] = EVENTLIRCD_EVKEY_TYPE_KEY,
+	[ BTN_TRIGGER_HAPPY    ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY2   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY3   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY4   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY5   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY6   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY7   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY8   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY9   ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY10  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY11  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY12  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY13  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY14  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY15  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY16  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY17  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY18  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY19  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY20  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY21  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY22  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY23  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY24  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY25  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY26  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY27  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY28  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY29  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY30  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY31  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY32  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY33  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY34  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY35  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY36  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY37  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY38  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY39  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+	[ BTN_TRIGGER_HAPPY40  ] = EVENTLIRCD_EVKEY_TYPE_BTN,
+};
+#endif
